packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-cognito-app-client-secret.ts

```diff
@@ -0,0 +1,51 @@
+import { $TSContext, AmplifyCategories, stateManager } from '@aws-amplify/amplify-cli-core';
+import { getAuthResourceName } from '../../../utils/getAuthResourceName';
+import { AuthInputState } from '../auth-inputs-manager/auth-input-state';
+import { MetaOutput } from '../import/types';
+import { getAppClientSecretViaSdk } from './get-app-client-secret-sdk';
+import { projectHasAuth } from './project-has-auth';
+
+/**
+ *
+ * returns app client secret if user pool exists
+ */
+const generateAppClientSecret = async (
+  context: $TSContext,
+  authResourceName: string,
+  clientId: string,
+  userPoolId: string,
+): Promise<string | undefined> => {
+  const authState = new AuthInputState(context, authResourceName);
+  if (authState.cliInputFileExists()) {
+    const { cognitoConfig } = authState.getCLIInputPayload();
+    if (cognitoConfig.userpoolClientGenerateSecret) {
+      const appClientSecret = await getAppClientSecretViaSdk(context, userPoolId, clientId);
```

Comment:
```suggestion
      const appClientSecret = await getAppClientSecret(context, userPoolId, clientId);
```

1. isn't "how" implementation detail? would caller change if we substituted SDK call with something else?
2. should this be called `generateAppClientSecret` ?